presto-main/src/test/java/com/facebook/presto/operator/scalar/TestQuantileDigestFunctions.java

```diff
@@ -1,5 +1,143 @@
+import static org.testng.Assert.assertEquals;
+
 public class TestQuantileDigestFunctions
+        extends AbstractTestFunctions
 {
+    private TypeRegistry typeRegistry;
+
+    @BeforeClass
+    public final void initTypeRegistry()
+    {
+        typeRegistry = new TypeRegistry();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public final void destroyTypeRegistry()
+    {
+        typeRegistry = null;
+    }
+
+    @Test
+    public void testNullQuantileDigestGetValueAtQuantile()
+    {
+        functionAssertions.assertFunction("value_at_quantile(CAST(NULL AS qdigest(bigint)), 0.3)", BIGINT, null);
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void testGetValueAtQuantileOverOne()
+    {
+        QuantileDigest qdigest = new QuantileDigest(1);
+        functionAssertions.assertFunction(format("value_at_quantile(CAST(X'%s' AS qdigest(bigint)), 1.5)", serialize(qdigest)),
+                BIGINT,
+                null);
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void testGetValueAtQuantileBelowZero()
+    {
+        QuantileDigest qdigest = new QuantileDigest(1);
+        functionAssertions.assertFunction(format("value_at_quantile(CAST(X'%s' AS qdigest(bigint)), -0.2)", serialize(qdigest)),
+                BIGINT,
+                null);
+    }
+
+    @Test
+    public void testValueAtQuantileBigint()
+    {
+        QuantileDigest qdigest = new QuantileDigest(1);
+        addAll(qdigest, asList(0L, 1L, 2L, 3L, 4L, 5L, 6L, 7L, 8L, 9L));
+
+        functionAssertions.assertFunction(format("value_at_quantile(CAST(X'%s' AS qdigest(bigint)), 0.5)", serialize(qdigest)),
+                BIGINT,
+                5L);
+    }
+
+    @Test
+    public void testScale()
+    {
+        QuantileDigest qdigest = new QuantileDigest(1);
+        addAll(qdigest, asList(0L, 1L, 2L, 3L, 4L, 5L, 6L, 7L, 8L, 9L));
+
+        // Before scaling.
+        assertEquals(qdigest.getHistogram(asList(0L, 1L, 2L, 3L, 4L, 5L, 6L, 7L, 8L, 9L, 10L)),
+                asList(new QuantileDigest.Bucket(0, Double.NaN),
+                        new QuantileDigest.Bucket(1, 0),
+                        new QuantileDigest.Bucket(1, 1),
+                        new QuantileDigest.Bucket(1, 2),
+                        new QuantileDigest.Bucket(1, 3),
+                        new QuantileDigest.Bucket(1, 4),
+                        new QuantileDigest.Bucket(1, 5),
+                        new QuantileDigest.Bucket(1, 6),
+                        new QuantileDigest.Bucket(1, 7),
+                        new QuantileDigest.Bucket(1, 8),
+                        new QuantileDigest.Bucket(1, 9)));
+
+        // Scale up.
+        SqlVarbinary sqlVarbinary = ((SqlVarbinary) functionAssertions.selectStringValue(format("scale_qdigest(CAST(X'%s' AS qdigest(bigint)), 2)", serialize(qdigest)),
+                QDIGEST.createType(typeRegistry, ImmutableList.of(TypeParameter.of(BIGINT)))));
```

Comment:
It might make sense to move this type creation to a before/after method (where the type manager is currently created), or move it all into the first line of the test and reused here and below